crates/sim2h/src/lib.rs

```diff
@@ -134,21 +604,28 @@ impl Sim2h {
 
         let (wss_send, wss_recv) = crossbeam_channel::unbounded();
         let (msg_send, msg_recv) = crossbeam_channel::unbounded();
-
+        let (tp_send, tp_recv) = crossbeam_channel::unbounded();
+        let state = Arc::new(RwLock::new(Sim2hState {
+            crypto: crypto.box_clone(),
+            connection_states: HashMap::new(),
+            open_connections: HashMap::new(),
+            spaces: HashMap::new(),
+        }));
         let mut sim2h = Sim2h {
             crypto,
+            state,
             bound_uri: None,
-            connection_states: RwLock::new(HashMap::new()),
-            spaces: HashMap::new(),
             pool,
             wss_recv,
             msg_send,
             msg_recv,
-            open_connections: HashMap::new(),
             num_ticks: 0,
             rrdht_arc_radius_recalc: std::time::Instant::now(),
             missing_aspects_resync: std::time::Instant::now(),
             dht_algorithm: DhtAlgorithm::FullSync,
+            threadpool: ThreadPool::new(num_cpus::get()),
```

Comment:
I don't think we really want two thread pools, right? We should just pick one of them to use.